Mathlib/Tactic/Basic.lean

```diff
@@ -124,3 +124,37 @@ elab (name := clearAuxDecl) "clear_aux_decl" : tactic => withMainContext do
     if ldec.isAuxDecl then
       g ← g.tryClear ldec.fvarId
   replaceMainGoal [g]
+
+/-- Clears the value of the local definition `fvarId`. Ensures that the resulting goal state
+is still type correct. Throws an error if it is a local hypothesis without a value. -/
+def _root_.Lean.MVarId.clearValue (mvarId : MVarId) (fvarId : FVarId) : MetaM MVarId := do
+  mvarId.checkNotAssigned `clear_value
+  let (xs, mvarId') ← mvarId.revert #[fvarId] true
+  mvarId'.withContext do
+    let numReverted := xs.size
+    let tgt ← mvarId'.getType
+    unless tgt.isLet do
+      mvarId.withContext <|
+        throwTacticEx `clear_value mvarId m!"{Expr.fvar fvarId} is not a local definition"
+    let tgt' := Expr.forallE tgt.letName! tgt.letType! tgt.letBody! .default
+    unless ← isTypeCorrect tgt' do
+      mvarId.withContext <|
+        throwTacticEx `clear_value mvarId
+          m!"cannot clear {Expr.fvar fvarId}, the resulting context is not type correct"
+    let mvarId'' ← mkFreshExprMVar tgt'
```

Comment:
should this be synthetic opaque? (probably not, but just checking)

Reply:
Yeah, I was wondering about this. I figured that `introNP` would be responsible for creating the final metavariable so it didn't matter, but maybe out of principle it should be synthetic opaque?